File: Data Structures & Algorithms/missing-number/submission-3.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        
        maxVal = max(nums)
        ans = 0

        for i in range(1,maxVal + 1):
            ans ^= i

        for num in nums:
            ans ^= num
        
        if ans == 0 and 0 in nums:
            return len(nums)
        else:
            return ans

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for(int num:nums){
            if (!set.contains(num)){
                set.add(num);
            }
        }
        List<Integer> start = new ArrayList<>();
        for(int num:nums){
            if(!set.contains(num - 1)){
                start.add(num);
            }
        }
        int res = 0;
        for(int i = 0;i < start.size();i++){
            int length = 0;
            while(set.contains(start.get(i) + length)){
                length++;
            }
            res = Math.max(res,length);
        }
        
        // System.out.println(start);
        return res;
    }
}
